File: fizzbuzz.r
library(collections) 
dict <- ordered_dict(keys=list(3, 5), items=list("Fizz", "Buzz"))
output <- ''
for(i in 1:100) {
  output <- ''
  for (key in dict$keys()) {
    if (i %% key == 0) {
      output <- paste(output, dict$get(key, ''), sep = "")
    }
  }
  if (output == "") {
    output <- toString(i)
  }
  print(output)
}

File: fizzbuzz.rs
use std::collections::BTreeMap;

fn main() {
    let mut dict = BTreeMap::new();
    let mut output = "".to_string();
    dict.insert(3, "Fizz");
    dict.insert(5, "Buzz");
    for i in 1..=100 {
        output.clear();
        for (key, value) in &dict {
            if i % key == 0 {
                output.push_str(value);
            }
        }
        if output.is_empty() {
            output = i.to_string();
        }
        println!("{}", output)
    }
}